### ISSUE TITLE
Exceptions with multi-threading

### ISSUE THREAD

**Issue by alfkonee** [axuno/SmartFormat#283]
Hi! Thanks for reporting this bug! 

Please fill in the relevant information from this template so that we can help you best.

**Smart.Format version**: 3.0.0

**Framework version**:  net6.0

- Please provide a working source code example to reproduce the bug
- What is the current result?
Exception then Memory and CPU Leak
![SMART FORMAT LEAK](https://user-images.githubusercontent.com/3909521/168749618-362a89e6-3cf6-4b34-81b0-020cdb7563a0.jpeg)

- What is the expected result?
No Exceptions in High Traffic Scenarios with Date Parsing
- Please post full exception details if applicable 
``` 
ValueFactory attempted to access the Value property of this instance.
System.Lazy`1 in ViaFactory in /usr/share/dotnet/shared/Microsoft.NETCore.App/6.0.5/System.Private.CoreLib.dll
System.Lazy`1 in CreateValue in /usr/share/dotnet/shared/Microsoft.NETCore.App/6.0.5/System.Private.CoreLib.dll
SmartFormat.Pooling.SmartPools.LiteralTextPool in get_Instance in /usr/share/dotnet/shared/Microsoft.NETCore.App/6.0.5/System.Private.CoreLib.dll
SmartFormat.Core.Parsing.Parser in ParseFormat in /usr/share/dotnet/shared/Microsoft.NETCore.App/6.0.5/System.Private.CoreLib.dll
SmartFormat.SmartFormatter in Format in /usr/share/dotnet/shared/Microsoft.NETCore.App/6.0.5/System.Private.CoreLib.dll
...
```
- Did you find a workaround? yes revert to v2.7.x
- Is there a version in which it worked? 2.7.2
- Can you help us by writing an unit test? NO please

**Comment by axunonb** [comment]
Without a working example it's hard to tell what went wrong.
My guess are 2 things:
1.  As multi-threading  or `async` seems to be used, `SmartSettings.IsThreadSafeMode = true` must be set. The default is `false`.
2. `Smart.Format(...)` methods can't be used in this case, as they use a static `SmartFormatter` instance. A new `SmartFormatter` must be created for each thread.

See the [Wiki](https://github.com/axuno/SmartFormat/wiki/Async-and-Thread-Safety) and the hint in the [README](https://github.com/axuno/SmartFormat) for details.

**Comment by alfkonee** [comment]
Thanks for the info about the thread safety I guess it's a 3.0 Feature that was added Right??

Yes I'm working in a Highly Multi threaded Environment will apply the fix and feedback the performance updates

**Comment by axunonb** [comment]
We just published release [v3.1.0](https://github.com/axuno/SmartFormat/releases/tag/v3.1.0):
* `SmartSettings.IsThreadSafeMode = true` is the default
* Static `Smart` methods like `Smart.Format(...)` can be used also with `async` / multi-threading
